### DISCUSSION TITLE
Tesla-History / error recovery

### DISCUSSION THREAD

**Discussion by youzer-name** [jasonacox/Powerwall-Dashboard#356] [General]
@mcbirse I've been having some "hiccups" on my Raspberry Pi that hosts Powerwall Dashboard, and I've had to restart the docker daemon a few times over the past few days to fix things. On a few occasions the tesla-history container has thrown this error:

>`
ERROR: Failed to execute InfluxDB query: SELECT * FROM autogen.http WHERE source='cloud' AND time >= '2023-09-21T23:51:41+00:00' AND time <= '2023-09-22T01:51:41+00:00'; HTTPConnectionPool(host='pw-influxdb', port=8086): Max retries exceeded with url: /query?q=SELECT+%2A+FROM+autogen.http+WHERE+source%3D%27cloud%27+AND+time+%3E%3D+%272023-09-21T23%3A51%3A41%2B00%3A00%27+AND+time+%3C%3D+%272023-09-22T01%3A51%3A41%2B00%3A00%27&db=solaronly (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xb59ed130>: Failed to establish a new connection: [Errno 111] Connection refused'))... Fix and restart.
Clearing InfluxDB data
`

I'm not sure if there is already some retry logic going on before it throws the error, but I don't see anything in the logs to indicate there is. Would it be a good idea to have the script retry in case the issue is temporary/intermittent?

It isn't a huge problem as-is, since I have monitoring set up to notify me if there is no new data going into the database,. When the data goes stale I just restart the container and manually run the script for --today and/or --yesterday to fill in any gaps. But it would be nice if the script could recover automatically.

**Comment by mcbirse** [comment]
@youzer-name - amazing timing on your post, as I am currently right now working on a new script based on the tesla-history and had come across this potential issue...

I am adding a "retry" option for when InfluxDB is unavailable or other errors occur, that should address this.

I am also considering/pondering that perhaps the docker-compose config for influxdb should have a "healthcheck" clause added, so that any service that "depends_on" influxdb will have a delayed startup or possibly restart if it becomes unhealthy. Still testing... however I think it will be better to have a retry in the tesla-history script itself.

I will roll my changes/improvements from the new script into the tesla-history as well, once I have sorted this out.

**Reply by mcbirse** [reply]
Hi @youzer-name - In the latest build of the tesla-history docker container I added some retry handling logic, which I'm hoping should fix the error you were experiencing.

Let me know how it goes when you have a chance to update.

Guessing you probably have a custom setup, so you would need to manually update your `powerwall.yml` (or equivalent) docker compose file, and update the build version to `0.1.3`, i.e.

```yaml
    tesla-history:
        image: jasonacox/tesla-history:0.1.3
```

**Reply by youzer-name** [reply]
@mcbirse - Thanks, I just updated. I haven't had dropouts with the database connectivity recently, but I'll check the logs occasionally and see if any recoveries get logged.
